src/app/dashboard/layout.tsx

```diff
@@ -17,8 +17,9 @@ Modal.setAppElement('#modals');
 const queryClient = new QueryClient({
   defaultOptions: {
     queries: {
+      retry: false,
       staleTime: Infinity,
-      refetchOnMount: false,
+      refetchOnMount: true,
```

Comment:
okay so I just discovered this is key. Instead of having to mark all invalidations with `refetchType: all` we can just invalidate (so the active ones refetch) and the non active ones will refetch once you remount the component that uses it. Quite neat and avoids doing to much computation on changes.